### DISCUSSION TITLE
Bookmark on shake to reset sleep timer

### DISCUSSION THREAD

**Discussion by nookits** [PaulWoitaschek/Voice#1895] [Ideas]
I use the sleep timer to listen to my audio books before bed. I don't like loosing my position in the book when I eventually fall asleep. I use the shake to reset feature to continue listening. None the less I find the next morning I don't know how far back to seek to find my place. This is because a bookmark is only created for when the sleep timer is first set. 

It would be nice to have a bookmark created when shaking to reset sleep timer. 

I have tried setting the sleep timer to a small value so that I wouldn't need to seek back very far. However, this hinders my falling to sleep. 

It is possible that a large number of bookmarks could be created if this feature is implemented. Perhaps it is worth while adding another option when setting the sleep timer.

**Comment by bjohas** [comment]
Extending #1093, this might be able to be addressed by adding
```
      lifecycleScope.launch {
        val book = bookRepo.get(args.getBookId(NI_BOOK_ID)!!) ?: return@launch
        bookmarkRepo.addBookmarkAtBookPosition(
          book = book,
          setBySleepTimer = true,
          title = null
        )
      }
```
In the places marked 'HERE'
```
  private fun start() {
    Logger.i("Starting sleepTimer. Pause in $sleepTime.")
    leftSleepTime = sleepTime
    sleepJob?.cancel()
    sleepJob = scope.launch {
      startSleepTimerCountdown()
      val shakeToResetTime = 30.seconds
      Logger.d("Wait for $shakeToResetTime for a shake")
      withTimeout(shakeToResetTime) {
        shakeDetector.detect()
        Logger.i("Shake detected. Reset sleep time")
        playerController.play()
        // HERE
        start()
      }
      if (playStateManager.playState == Playing) {
        Logger.i("Sleep timer finished, player is playing, restart timer.")
        // HERE
        start()
      }
      Logger.i("exiting")
    }
  }
```
